### ISSUE TITLE
Inconsistent rendering of Button widget

### ISSUE THREAD

**Issue by pbrown12303** [fyne-io/fyne#4243]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

The rendering of the button widget is different before the button is first clicked and after the button has been clicked. Specifically, the box used for the animation of the button has no color and a size of (0,0) prior to the first click and a size that is the same as the button and a color of transparent after the first click. This creates a problem for tests that use the test.AssertObjectRendersTo... and test.AssertRendersTo... functions.

### How to reproduce

Run the attached test code. The first run will fail because there is nothing in testdata, but after you move the two .xml files from failed to testdata it will succeed. Now compare beforeClick.xml to afterClick.xml. The rendering difference is clear.

### Screenshots

![image](https://github.com/fyne-io/fyne/assets/18626700/9ac5d940-0c12-4fc3-a518-806aee97861d)

### Example code

```go
package widget

import (
	"testing"
	"time"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/test"
	"fyne.io/fyne/v2/widget"
)

func TestButtonBg(t *testing.T) {
	button := widget.NewButton("ClickMe", nil)
	time.Sleep(100 * time.Millisecond)
	test.AssertObjectRendersToMarkup(t, "beforeClick.xml", button)
	button.Tapped(&fyne.PointEvent{})
	time.Sleep(100 * time.Millisecond)
	test.AssertObjectRendersToMarkup(t, "afterClick.xml", button)
}

```

### Fyne version

2,4,0

### Go compiler version

1.21.1

### Operating system and version

Windows 10

**Comment by andydotxyz** [comment]
Thanks. Feel free to open a PR with that change

**Comment by andydotxyz** [comment]
Landed on develop a while back and will be in 2.4.2
